### ISSUE TITLE
OpenOSRS support?

### ISSUE THREAD

**Issue by f0rmatme** [GigiaJ/RSB#41]
https://github.com/open-osrs/runelite

has more of a open API and you can access the mixins

**Comment by GigiaJ** [comment]
Alright so, this answer will be somewhat wordy.
I'd imagine, even now, one _could_ plug the bot api into OpenOSRS and it would work.
Now if not, could I make the existing bot API I have work with this? Probably. BUT, I have actively gone out of my way to use the stock RuneLite build. It is more popular AND it lacks the deobfuscator.

I actually could likely build the deobfuscator and what you call the _mixins_ personally without using the iteration they pulled from RuneLite. While I haven't done any Java reversing I hear it is fantastically easier with reflection and all. The deob is what reverses the game code and allows the mixins to actually have an effect on the way the game plays. In non-java games this would be done by opening IDA Pro, ripping the game binary apart and identifying functions from the x86 assembly, creating a signature that you can re-find the function next update in the binary (which if your signature is bad you are SOL), then take the identified functions memory address, create a hook in your code to modify the x86 assembly of the function so it performs the way you want, and finally inject this into your game. (Add in the loads of anti-cheat you usually need to work around) I'd imagine Java reversing works similar, but instead you don't even need to inject a dll into the game, but you can recompile all the code and run it. After de-obfuscating of course. (Definitely a little more complex, but not Java reversing doesn't seem too bad.)

That said I have no real reason or desire to use mixins. If I can't get a certain function because it isn't implemented I'll give it a go mathematically with the data I do have. The goal here is that using only what RuneLite provides we really waive off any DMCA or action because we do not have any source code up. (Jagex had a problem with RuneLite even having their deob up)
We want to prove that third party clients are a problem, and offer an automation api that abuses only what the most popular API offers.

TL;DR from experience I could craft the mixins myself given some time, but I don't actually want them. I want to prove RuneLite is a bad idea just off of its existence and that people should feel free to abuse that. (I actually don't play OSRS at all). 

If you can give me a decent reason as to why the bot API _needs_ to support it I will definitely consider it.

Really sorry if any of this comes off mean or pretentious. I try my best to be positive and respond positively. Don't think I'm just dismissing the idea please.

**Comment by f0rmatme** [comment]
Well OpenOSRS has nice features that would complement the bot like extended click distance and more. And a larger API to use with the bot. It might already work ill check sooner or later.

**Comment by f0rmatme** [comment]
Did not really explain this well, but simply put OpenOSRS destroys RuneLite when it comes to features.
Here is one example: https://github.com/open-osrs/runelite/pull/1687

This API change is just one pull request.

OpenOSRS has nice features that would work well with RSB and it keeps getting expainded. there are some really cool  changes coming " https://prnt.sc/rucnjz " that will make runelite jelly.
